src/app/rundown/components/piece/piece.component.ts

```diff
@@ -11,5 +11,76 @@ export class PieceComponent {
   @Input()
   public piece: Piece
 
-  constructor() { }
+  @Input()
+  public pixelsPerSecond: number
+
+  @Input()
+  public partDuration: number
+
+  @Input()
+  public expectedPartDuration: number
+
+  public getPieceWidthInPixels(): string {
+    const width = this.getPieceWidth()
+    return `${width}px`
+  }
+
+  private getPieceWidth(): number {
+    const duration  = this.piece.duration ?? (this.partDuration - this.piece.start)
+    return Math.floor(this.pixelsPerSecond * duration / 1000)
+  }
+
+  public getOffsetInPixels(): string {
+    const offset = this.getOffset()
+    return `${offset}px`
+  }
+
+  private getOffset(): number {
+    return Math.floor(this.pixelsPerSecond * (this.piece.start ?? 0) / 1000)
+  }
+
+  public getPieceTypeModifierClass(): string {
+    const pieceType = this.getPieceType()
+    return `piece--${pieceType}`
+  }
+
+  // TODO: Use either piece.type or source layer types for determining style.
+  public getPieceType(): string {
+    const layerType = this.piece.layer.replace(/^studio\d+_/, '')
+    switch (layerType) {
```

Comment:
Piece.type was intented to be used for this

Reply:
Yes, but it is not yet populated in the backend. And we focused on the sprint-goal :wink: